### PR TITLE
Fix fluids for CUDA_ERROR_INVALID_PTX and enable lv-cuda

### PR DESCRIPTION
**Pull request by jedbrown**
Thanks @nbeams for help debugging.

### CODE REVIEW

File: examples/fluids/qfunctions/advection2d.h [hunk 1]
```diff
@@ -36,6 +36,8 @@ struct AdvectionContext_ {
   int stabilization; // See StabilizationType: 0=none, 1=SU, 2=SUPG
 };
 
+CEED_QFUNCTION_HELPER CeedScalar Square(CeedScalar x) { return x*x; }
```
Comment (jeremylt) [inline examples/fluids/qfunctions/advection2d.h:39]: Probably could put this in a common header shared between the QFunctions with an include via rpath

Comment (jedbrown) [reply examples/fluids/qfunctions/advection2d.h:39]: Yeah, I didn't want to create that coupling just yet, but I envision a common header with utilities for scalars and 3x3 matrices.

**Comment by jeremylt** [comment]
There was some error in CI on LV that was before the part of the script touched by this PR, so I'm re-running.

**Comment by jedbrown** [comment]
Was your PETSc built for sm_61? Maybe need to export `CUDA_VISIBLE_DEVICES` when building PETSc, or pass `--with-cuda-arch=75`.

> Kokkos::Cuda::initialize ERROR: running kernels compiled for compute capability 6.1 on device with compute capability 7.5 is not supported by CUDA!

**Comment by jeremylt** [comment]
> Was your PETSc built for sm_61? Maybe need to export `CUDA_VISIBLE_DEVICES` when building PETSc, or pass `--with-cuda-arch=75`.
> 
> > Kokkos::Cuda::initialize ERROR: running kernels compiled for compute capability 6.1 on device with compute capability 7.5 is not supported by CUDA!

huh, I just used the same configure options that previously worked. I'll try adding more flags.

**Comment by jeremylt** [comment]
Fully rebulit PETSc on LV. It works locally, so I'm letting CI try again

**Comment by jeremylt** [comment]
ugh, I'm probably testing on a different device when I log in to try

```
Kokkos::Cuda::initialize ERROR: running kernels compiled for compute capability 6.1 on device with compute capability 7.5 is not supported by CUDA!
```

Still no idea why PETSc seems to be ignoring `--with-cuda-arch=75`. That's super convenient

**Comment by jeremylt** [comment]
Ok, I seem to have rebuilt sucessfully. I'm letting CI try again, though 3/4 of the runs have some random error in a random *t*` series test, so its slow going.

**Comment by jedbrown** [comment]
Maybe we should run a judicious small subset of the `t*` in CI, perhaps with lower `NPROC_GPU`. If fluids and solids pass on CUDA and full pipelines pass on ROCm, chances are the unit tests also pass on CUDA.

**Comment by jedbrown** [comment]
I have this in my `~/.bash_profile`
```bash
export CUDA_VISIBLE_DEVICES=GPU-c4529365-8229-f689-b43d-ccd7f1677079
. /opt/rh/gcc-toolset-11/enable
export PETSC_DIR=$HOME/petsc PETSC_ARCH=ompi-cuda-arch75-g
export PATH=/usr/lib64/openmpi/bin:$PATH
```

**Comment by jeremylt** [comment]
The strange thing is that sometimes `t000` fails and everything else passes, sometimes a random `t3*` or `t5*` fails. I haven't been able to pin down exactly why

---

Yeah, I set the `CUDA_VISIBLE_DEVICES` and that seemed to fix the PETSc build (I think)

**Comment by jeremylt** [comment]
This is the intermittent error
```
libgcov profiling error:/home/gitlab-runner/builds/st7hbLD-/0/libceed/libCEED/build/t004-ceed-f.gcda:overwriting an existing profile data with a different timestamp
```
I really don't see why this should ever happen though. The directory is cleared out before we run junit

**Comment by jeremylt** [comment]
This fix is now working, so I merged. I still need to dig more into why `lcov` isn't working on LV though

**Pull request merged**